### ISSUE TITLE
Infinite respawn loop with 0.46.0 (Worker exited (code: 2))

### ISSUE THREAD

**Issue by sebcode** [facebook/flow#4021]
Steps to reproduce:

```
mkdir test && cd test
yarn add flow-bin@0.46.0 fbjs@0.8.12
    
# Create .flowconfig with contents:
[options]
module.name_mapper='^\(core\|test\)\(\/\)?\(.*\)$' -> '<PROJECT_ROOT>/app/\1\2\3'
    
node_modules/.bin/flow
```

Output:

```
$ node_modules/.bin/flow
Launching Flow server for /private/tmp/test
Spawned flow server (pid=33230)
Logs will go to /private/tmp/flow/zSprivatezStmpzStest.log
Launching Flow server for /private/tmp/test
Spawned flow server (pid=33255)
Logs will go to /private/tmp/flow/zSprivatezStmpzStest.log
Launching Flow server for /private/tmp/test
[ ... infinite respawn loop ... ]
```

```
$ tail -f /private/tmp/flow/zSprivatezStmpzStest.log
[2017-05-25 13:12:48] Initializing Server (This might take some time)
[2017-05-25 13:12:48] executable=/private/tmp/test/node_modules/flow-bin/flow-osx-v0.46.0/flow
[2017-05-25 13:12:48] version=0.46.0
[2017-05-25 13:12:48] Parsing
[2017-05-25 13:12:48] Building package heap
[2017-05-25 13:12:48] Loading libraries
Exception: Failure("Str.replace: reference to unmatched group")
Potential backtrace:
Worker exited (code: 2)
Subprocess(33486): fail 2Worker.Worker_exited_abnormally(2)
```

Expected output (with flow-bin 0.45.0):
```
$ node_modules/.bin/flow
Launching Flow server for /private/tmp/test
Spawned flow server (pid=33557)
Logs will go to /private/tmp/flow/zSprivatezStmpzStest.log
No errors!
```

**Comment by sebcode** [comment]
Seems to be fixed since 0.48.0.

**Comment by sdeleon28** [comment]
For the record, this happened to me because I had this malformed `module.name_mapper` entry:

```
module.name_mapper='<icons>' -> '<PROJECT_ROOT>/src/icons/\1'
```

Notice the `\1` in the replacement pattern has no corresponding group in the regex.
